File: 3sem/4lab/Banks/Exceptions/InvalidDepositInterestRateSelectorException.cs
namespace Banks.Exceptions;

public class InvalidDepositInterestRateSelectorException : BanksDomainException
{
    private InvalidDepositInterestRateSelectorException(string? message)
        : base(message) { }

    public static InvalidDepositInterestRateSelectorException OnGapsIntersection()
        => new InvalidDepositInterestRateSelectorException($"Money gaps must not intersect");

    public static InvalidDepositInterestRateSelectorException OnEmptyGapsList()
        => new InvalidDepositInterestRateSelectorException($"There must be at least one money gap");
}

File: 4sem/1lab/src/main/java/BanksRemake/Domain/Exceptions/InvalidDepositInterestRateSelectorException.java
package BanksRemake.Domain.Exceptions;

public class InvalidDepositInterestRateSelectorException extends BanksRemakeException {
    public InvalidDepositInterestRateSelectorException(String message) {
        super(message);
    }

    public static InvalidDepositInterestRateSelectorException onGapsIntersection() {
        return new InvalidDepositInterestRateSelectorException("Money gaps must not intersect");
    }

    public static InvalidDepositInterestRateSelectorException onEmptyGapsList() {
        return new InvalidDepositInterestRateSelectorException("There must be at least one money gap");
    }
}
